frontend/src/app/modules/user/user-preferences/user-preferences.component.ts

```diff
@@ -0,0 +1,140 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { BaseComponent } from '@core/base/base.component';
+import { LanguageService } from '@core/services/language.service';
+import { ToastrNotificationsService } from '@core/services/toastr-notifications.service';
+import { UserService } from '@core/services/user.service';
+import { ILanguage } from '@shared/models/language/language';
+import { IUser } from '@shared/models/user/user';
+import { INewUserPreferences } from '@shared/models/user-prefferences/new-user-preferences';
+import { IUserPreferences } from '@shared/models/user-prefferences/user-preferences';
+import { fontSizeMaxValue, tabSizeMaxValue } from '@shared/utils/validation/form-control-validator-options';
+import { getErrorMessage } from '@shared/utils/validation/validation-helper';
+import { takeUntil } from 'rxjs';
+
+import { findItemIdByName, MONACO_EDITOR_THEMES } from './user-preferences.utils';
+
+@Component({
+    selector: 'app-user-preferences',
+    templateUrl: './user-preferences.component.html',
+    styleUrls: ['./user-preferences.component.sass'],
+})
+export class UserPreferencesComponent extends BaseComponent implements OnInit {
+    userPreferenceForm: FormGroup = new FormGroup({
+        language: new FormControl(''),
+        theme: new FormControl(''),
+        tabSize: new FormControl(3, [Validators.max(tabSizeMaxValue)]),
+        fontSize: new FormControl(15, [Validators.max(fontSizeMaxValue)]),
+        wordWrap: new FormControl(false),
+        minimap: new FormControl(false),
+    });
+
+    user: IUser;
+
+    languagesNames: string[] = [];
+
+    isWordWrap: boolean;
+
+    private languages: ILanguage[] = [];
+
+    monacoThemes: string[] = MONACO_EDITOR_THEMES;
+
+    constructor(
+        private toastrService: ToastrNotificationsService,
+        private languageService: LanguageService,
+        private userervice: UserService,
+    ) { super(); }
+
+    ngOnInit(): void {
+        this.getLanguages();
+        this.loadData();
+    }
+
+    setPreferences() {
```

Comment:
and we need to save everything on click on Save (do not have a btn on every block)